sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs

```diff
@@ -1457,6 +1458,44 @@ public async Task GetPropertiesAsync_BlobSAS()
 
         }
 
+        [Test]
+        public async Task GetPropertiesAsync_BlobSasWithContentHeaders()
+        {
+            var containerName = GetNewContainerName();
+            var blobName = GetNewBlobName();
+            await using DisposingContainer test = await GetTestContainerAsync(containerName: containerName);
+
+            // Arrange
+            BlobBaseClient blob = await GetNewBlobClient(test.Container, blobName);
+
+            BlobSasBuilder blobSasBuilder = new BlobSasBuilder
+            {
+                BlobContainerName = containerName,
+                BlobName = blobName,
+                Protocol = SasProtocol.None,
+                StartsOn = Recording.UtcNow.AddHours(-1),
+                ExpiresOn = Recording.UtcNow.AddHours(1),
+                IPRange = new SasIPRange(IPAddress.None, IPAddress.None),
+                CacheControl = "|cache control?",
+                ContentDisposition = "|content disposition?",
+                ContentEncoding = "|content encoding?",
+                ContentLanguage = "|content language?",
+                ContentType = "|content type?"
+            };
+            blobSasBuilder.SetPermissions(
+                BlobSasPermissions.All);
+
+            BlobSasQueryParameters blobSasQueryParameters = blobSasBuilder.ToSasQueryParameters(GetNewSharedKeyCredentials());
+
+            BlockBlobClient sasBlob = InstrumentClient(new BlockBlobClient(new Uri($"{TestConfigDefault.BlobServiceEndpoint}/{containerName}/{blobName}?{blobSasQueryParameters}"), GetOptions()));
```

Comment:
Better to do something like the following instead of formatting Uri ourselves:
```
var blobUri = new BlobUriBuilder(new Uri(TestConfigDefault.BlobServiceEndpoint));
blobUri.BlobName = blobName;
blobUri.BlobContainerName = containerName;
blobUri.Sas = blobSasQueryParameters;
BlockBlobClient sasBlob = InstrumentClient(new BlockBlobClient(blobUri.ToUri(), GetOptions()));
```

Reply:
fixed.